### ISSUE TITLE
runit : sv stop doesn't work

### ISSUE THREAD

**Issue by hamelg** [graphite-project/docker-graphite-statsd#88]
It seems there is an issue with runit.

```
# sv start carbon-aggregator
ok: run: carbon: (pid 27255) 0s
# ps |egrep '(28032|carbon-aggregator)'
    8 root      0:00 runsv carbon-aggregator
28032 root      0:00 {run} /bin/sh ./run
28034 root      0:00 python3 /opt/graphite/bin/carbon-aggregator.py start --debug
28035 root      0:00 tee -a /var/log/carbon-aggregator.log
28245 root      0:00 egrep (28032|carbon-aggregator)
# sv stop carbon-aggregator
ok: down: carbon-aggregator: 0s, normally up
# ps |egrep '(28032|carbon-aggregator)'
    8 root      0:00 runsv carbon-aggregator
28034 root      0:00 python3 /opt/graphite/bin/carbon-aggregator.py start --debug
28035 root      0:00 tee -a /var/log/carbon-aggregator.log
28324 root      0:00 egrep (28032|carbon-aggregator)
```
It kills the shell but not the service.

**Comment by hamelg** [comment]
"sv stop" doesn't work because the service is not the direct child of runsv.

```
\_ runsv carbon
|   \_ /bin/sh ./run
|       \_ python3 /opt/graphite/bin/carbon-cache.py start --debug
|       \_ tee -a /var/log/carbon.log

# cat /etc/service/carbon/run 
#!/bin/sh

rm -f /opt/graphite/storage/carbon-cache-a.pid
exec python3 /opt/graphite/bin/carbon-cache.py start --debug 2>&1 | tee -a /var/log/carbon.log
```

In the run script, the exec builtin has no effect because the shell fork/exec another shell to process the pipe. The result is the service carbon doesn't receive SIGTERM when the service is stopping.

But, why you want to log twice : once in a file and once via the docker log driver ?

**Comment by deniszh** [comment]
> But, why you want to log twice : once in a file and once via the docker
log driver ?

Originally it was only file -
https://github.com/hopsoft/docker-graphite-statsd - but iirc somebody asked
(or created pr) for stdout too. But that’s not a problem here.
Initial script was running in Ubuntu, maybe exec behaves differently in
alpine because of musl or busybox...

**Comment by hamelg** [comment]
> Initial script was running in Ubuntu, maybe exec behaves differently in alpine because of musl or busybox...

You'll get the same behavior whatever the shell you use. It's how the runit works.
To fix that you can split the run script in 2 parts : 

```
# cat /etc/service/carbon/run 
#!/bin/sh
exec python3 /opt/graphite/bin/carbon-cache.py start --debug 2>&1
# cat /etc/service/carbon/log/run 
#!/bin/sh
exec tee -a /var/log/carbon.log


```

**Comment by deniszh** [comment]
Ah, right, we should do logging in runit way, indeed. OK, thanks, will fix it.

**Comment by deniszh** [comment]
Fixed in v1.1.5-8.
Thanks!
